src/Migrations.ts

```diff
@@ -0,0 +1,204 @@
+  process.cwd(),
+  DEFAULT_MIGRATIONS_DIR_NAME
+)
+
+const MIGRATION_FILE_REGEXP = /^(\d+).(.*?)\.sql$/
+const MIGRATION_FILE_UP_REGEXP = /^-- .*$/gm
+const MIGRATION_FILE_DOWN_REGEXP = /^--\s+?down\b/im
+const MIGRATION_FILE_REPLACE_COMMAND_REGEXP = /^-{3,}$/gim
+
+export class Migrations<Driver extends Database> {
+  db: Driver
+  config: MigrationParams
+
+  constructor (db: Driver, config?: MigrationParams) {
+    this.db = db
+    this.config = {
+      force: false,
+      table: DEFAULT_MIGRATIONS_DIR_NAME,
+      ...config,
+      migrationsPath: path.resolve(
+        config?.migrationsPath ?? DEFAULT_MIGRATIONS_DIR_PATH
+      )
+    }
+  }
+
+  /**
+   * Performs a database migration read operation.
+   *
+   * You can run up and down with `db.exec(migrations[0].up)` or `db.exec(migrations[0].down)`. Or with other commands that can run SQL code.
+   *
+   * @param migrationPath path to migration folder can be relative or absolute. If not found uses current working directory
+   * @returns IMigrate.MigrationData[]. List of objects with name, id of migrations and SQL code for up and down migration.
+   *
+   * Get the list of migrations, for example:
+   *  { id: 1, name: 'initial', filename: '001-initial.sql', up: ..., down: ... }
+   *  { id: 2, name: 'feature', filename: '002-feature.sql', up: ..., down: ... }
+   */
+  async readMigrations (): Promise<MigrationData[]> {
+    const migrationFiles = await this.getMigrationFiles()
+    if (!migrationFiles.length) {
+      throw new Error(
+        `No migration files found in '${this.config.migrationsPath}'.`
+      )
+    }
+
+    return Promise.all(
+      migrationFiles.map(migration => this.readMigrationData(migration))
+    )
+  }
+
+  // TODO: propose to make method GetInstance of the Migrations class, which will be async and we will call
+  // `downMigrations` and `upNewMigrations` functions in the migrate method. And user can call these
+  // `downMigrations` and `upNewMigrations` functions when ever he wants it
+  /**
+   * Performs a database migration.
+   *
+   * Migrates database schema to the latest version. Filename option is present
+   * in the the MigrationData if `readMigrations` function executes or user can put filename
+   * for each migration in the config.migrations
+   */
+  async migrate (): Promise<void> {
+    const table = this.config.table
+    const migrations = this.config.migrations ?? (await this.readMigrations())
+
+    // Create a database table for migrations meta data if it doesn't exist
+    await this.db.run(`CREATE TABLE IF NOT EXISTS "${table}" (
+      id   INTEGER PRIMARY KEY,
+      name TEXT    NOT NULL,
+      up   TEXT    NOT NULL,
+      down TEXT    NOT NULL
+    )`)
+
+    // Get the list of already applied migrations
+    let dbMigrations = await this.db.all(
+      `SELECT id, name, up, down FROM "${table}" ORDER BY id ASC`
+    )
+
+    await this.downMigrations(dbMigrations, migrations)
```

Comment:
Remove these two calls so this fn does one specific thing - just inits the tables